faculty_cli/cli.py

```diff
@@ -1239,3 +1241,33 @@ def ls(project, path):
             click.echo("/project{}/".format(item.path))
         else:
             click.echo("/project{}".format(item.path))
+
+
+@cli.group()
+def completion():
+    """Generate auto-completion scripts for faculty_cli."""
+    pass
+
+
+@completion.command()
+def bash():
```

Comment:
I'm happy to keep these explicit if you prefer, but did you consider having a single command that infers the shell by checking the `SHELL` environment variable?

Reply:
That's a nice idea. My preference would be to keep it explicit for the sake of simplicity. But I'm happy to use $SHELL if anyone has a strong preference for this.

Reply:
What I've seen in similar tools which works well is an optional argument to specify the shell, e.g. `faculty completions -s bash`, where the `-s` argument defaults to `$SHELL`.

Reply:
I don't have strong feeling on this.

Minor note: If keeping the shell as a required argument, the CLI help might be a bit better if you use a 'choice' argument instead of subcommands, e.g.:

```python
@click.command()
@click.argument("shell", type=click.Choice(["bash", "zsh", "fish"]))
def completion(shell):
    click.echo(shell)
```